cni/nnsgrpcclient/nnsgrpcclient_test.go

```diff
@@ -0,0 +1,117 @@
+package nnsgrpcclient
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-container-networking/test/nnsmockserver"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var mockserver *nnsmockserver.NnsMockServer
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	mockserver  = nnsmockserver.NewNnsMockServer()
+	go  mockserver.StartGrpcServer(nnsPort)
+	fmt.Printf("mock nns server started\n")
+}
+
+func teardown() {
+	mockserver.StopGrpcServer()
+	fmt.Printf("mock nns server stopped\n")
+}
+
+// CNI ADD to add container to network
+func TestAddContainerNetworking(t *testing.T) {
+
+	client := &NnsGrpcClient{}
+	if err := client.AddContainerNetworking(
+		 context.Background(),
+		"sf_8e9961f4-5b4f-4b3c-a9ae-c3294b0d9681",
+		"testnwspace"); err != nil {
+       t.Fatalf("TestAddContainerNetworking failed: %v", err)
+	}
+
+	fmt.Printf("TestAddContainerNetworking success\n")
+}
+
+// CNI DEL to delete container from network
+func TestDeleteContainerNetworking(t *testing.T) {
+
+	client := &NnsGrpcClient{}
+	if err := client.DeleteContainerNetworking(
+		context.Background(),
+		"sf_8e9961f4-5b4f-4b3c-a9ae-c3294b0d9681",
+		"testnwspace"); err != nil {
+		t.Fatalf("TestSetupContainer: %v", err)
+	}
+
+	fmt.Printf("TestDeleteContainerNetworking: success\n")
+}
+
+// CNI ADD to add container to network - failure case
+func TestAddContainerNetworkingFailure(t *testing.T) {
+
+	client := &NnsGrpcClient{}
+
+	var err error
+	if err = client.AddContainerNetworking(context.Background(), "testpod", "testnwspace"); err == nil {
+		t.Fatalf("TestAddContainerNetworkingFailure failed. Expected error but none returned\n")
+	}
+
+	if !strings.Contains(err.Error(), "Setup") {
```

Comment:
Checking the string representation of errors is brittle. It's better to have error types. In this case, does it matter what specific error was returned? Same for below

Reply:
This is not needed as such. I added it because same API in mock server returns result of both Add/Delete. So this extra check here makes sure of right operation was done on mock server.